### ISSUE TITLE
Deploying multiple Triton containers on a single GPU and unable to servce by GRPC at the same time

### ISSUE THREAD

**Issue by winstonhic** [triton-inference-server/server#5052]
**Description**

1. When multiple triton docker containers on a single GPU and model instances are ready to serve, grpc clients send multiple requests at the same time. After that, some containers will be crashed. 

2. Sometimes, triton server will take a long time to load model instance, or just suspend.

**Triton Information**
NVIDIA Release 21.10 (build 28453983)

Are you using the Triton container or did you build it yourself?
Triton container:
FROM nvcr.io/nvidia/tritonserver:21.10-py3

**To Reproduce**
1. To ensemble 3 model intances. (python_backend-> onnx -> python_backend)
2.  command: ["tritonserver", "--model-repository=/models", "--strict-model-config=false", "--grpc-infer-allocation-pool-size=1",                              "--log-verbose=1"]
3. use docker-compose up --scale inference_server=2 -d, with shm_size: 1g and port mapping : 
ports:
      - 8000-8001:8000
      - 8100-8101:8001
      - 8200-8201:8002
4. To inference by two grpc clients each sends 100 sequent requests to different port(e.g. to 8100 and 8101) at the same time. 
5. The container would randomly be stopped.  The log of triton serveris : 

 0# 0x00005622E45F58A9 in tritonserver
 1# 0x00007F644CC66210 in /usr/lib/x86_64-linux-gnu/libc.so.6
 2# 0x00007F64108E900C in /opt/tritonserver/backends/python/libtriton_python.so
 3# TRITONBACKEND_ModelInstanceExecute in /opt/tritonserver/backends/python/libtriton_python.so
 4# 0x00007F644D7F283A in /opt/tritonserver/bin/../lib/libtritonserver.so
 5# 0x00007F644D7F304D in /opt/tritonserver/bin/../lib/libtritonserver.so
 6# 0x00007F644D6A7801 in /opt/tritonserver/bin/../lib/libtritonserver.so
 7# 0x00007F644D7ECDC7 in /opt/tritonserver/bin/../lib/libtritonserver.so
 8# 0x00007F644D054DE4 in /usr/lib/x86_64-linux-gnu/libstdc++.so.6
 9# 0x00007F644D4D2609 in /usr/lib/x86_64-linux-gnu/libpthread.so.0
10# clone in /usr/lib/x86_64-linux-gnu/libc.so.6

6. But if only using one grpc client sent 100 requests to one grpc port(e.g. 8100 or 8101) , triton server can complete all of the inferences.

**Expected behavior**
Multipe triton containers with multipe model instances can be deployed on a single GPU and serve grpc simultaneously.

**Comment by Tabrizian** [comment]
@winstonhic Can you try a more recent Triton version and see whether you observe the same issue or not?

**Comment by dyastremsky** [comment]
Closing issue due to lack of activitity. If you need follow up, please let us know and we can reopen the issue.

**Comment by winstonhic** [comment]
I recently tried version 22.11, and I got the same issue, 
Here is my docker-compose.yml:
```
version: "3.9"

services:
  inference_server:
    image: nvcr.io/nvidia/tritonserver:22.11-py3
    scale: 5
    volumes:
      - ./models/triton:/models
     command: ["tritonserver", "--model-repository=/models",
                              "--grpc-infer-allocation-pool-size=16",
                              ]
    shm_size: 4g
    ulimits:
      memlock: -1
      stack: 67108864
    ipc: host
    network_mode: host
    
    runtime: nvidia
    deploy:
      resources:
        reservations:
          devices:
          - driver: nvidia
            capabilities: [gpu]
```

And in python-backend model, we installed python3-opencv and import cv2

The processes of every container were stuck with loading the python_backend model. The log is 
```
== Triton Inference Server ==
=============================

NVIDIA Release 22.11 (build 48581223)
Triton Server Version 2.28.0

Copyright (c) 2018-2022, NVIDIA CORPORATION & AFFILIATES.  All rights reserved.

Various files include modifications (c) NVIDIA CORPORATION & AFFILIATES.  All rights reserved.

This container image and its contents are governed by the NVIDIA Deep Learning Container License.
By pulling and using the container, you accept the terms and conditions of this license:
https://developer.nvidia.com/ngc/nvidia-deep-learning-container-license

WARNING: The NVIDIA Driver was not detected.  GPU functionality will not be available.
   Use the NVIDIA Container Toolkit to start this container with GPU support; see
   https://docs.nvidia.com/datacenter/cloud-native/ .

W0113 06:37:14.069875 1 pinned_memory_manager.cc:236] Unable to allocate pinned system memory, pinned memory pool will not be available: CUDA driver version is insufficient for CUDA runtime version
I0113 06:37:14.069946 1 cuda_memory_manager.cc:115] CUDA memory pool disabled
I0113 06:37:14.071787 1 model_lifecycle.cc:459] loading: image_prep:1
```

**Comment by Tabrizian** [comment]
@winstonhic @kthui has fixed a number of issues related to Python backend hangs [here ](https://github.com/triton-inference-server/python_backend/commit/32974481c4875ad22bf318dcb2b80e316cdf5065)and [here](https://github.com/triton-inference-server/python_backend/commit/00468e5f9d4d61c4b2fabf941b87424291e81e36). They will be included in 23.02 release. If they still didn't resolve the hang, please open a new issue.
